asset-transfer-basic/chaincode-external/README.md

```diff
@@ -47,37 +45,106 @@ tar cfz code.tar.gz connection.json
 
 ```
-peer lifecycle chaincode queryinstalled --peerAddresses {PEER_ADDRESS} 
+setGlobals 1
+../bin/peer lifecycle chaincode install ../asset-transfer-basic/chaincode-external/asset-transfer-basic-external.tgz
 ```
 
-Edit the `chaincode.env` file to configure the `CHAINCODE_ID` variable with obtained chaincode package-id.
+setGlobals simply defines a bunch of environment variables suitable to act as one organization or another, org1 or org2.
+
+Install it on org2:
+
+```
+setGlobals 2
+../bin/peer lifecycle chaincode install ../asset-transfer-basic/chaincode-external/asset-transfer-basic-external.tgz
+```
+
+This will output the chaincode pakage identifier such as `basic_1.0:0262396ccaffaa2174bc09f750f742319c4f14d60b16334d2c8921b6842c090c` that you will need to use in the following commands.
+
+If needed, you can query the installed chaincode to get chaincode its package-id:
+
+```
+setGlobals 1
+../bin/peer lifecycle chaincode queryinstalled --peerAddresses localhost:7051 --tlsRootCertFiles organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
+```
+
+Edit the `chaincode.env` file in the `fabric-samples/asset-transfer-basic/chaincode-external` directory as necessary to set the `CHAINCODE_ID` variable to the chaincode package-id obtained above.
+
 
 ## Running the Asset-Transfer-Basic external service
 
-To run the service in a container, build a Asset-Transfer-Basic docker image:
+To run the service in a container, from a different terminal, build an Asset-Transfer-Basic docker image, using the supplied `Dockerfile`, using the following command in the `fabric-samples/asset-transfer-basic/chaincode-external` directory:
 
 ```
 docker build -t hyperledger/asset-transfer-basic .
 ```
 
-Edit the `chaincode.env` file to configure the `CHAINCODE_ID` variable before starting a Asset-Transfer-Basic container using the following command:
+Then, start the Asset-Transfer-Basic service:
 
 ```
 docker run -it --rm --name asset-transfer-basic.org1.example.com --hostname asset-transfer-basic.org1.example.com --env-file chaincode.env --network=net_test hyperledger/asset-transfer-basic
 ```
 
-## Starting the Asset-Transfer-Basic external service
+This will start the container and start the external chaincode service within it.
+
+## Finish installing the Asset-Transfer-Basic external chaincode
+
+Finishing the installation of the chaincode on the test network can be done from the terminal you started the network from with the following commands (make sure the package-id is set to the right one):
+
+```
+setGlobals 2
+../bin/peer lifecycle chaincode approveformyorg -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile $PWD/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem --channelID mychannel --name basic --version 1.0 --package-id basic_1.0:0262396ccaffaa2174bc09f750f742319c4f14d60b16334d2c8921b6842c090c --sequence 1
```

Comment:
Maybe use a variable to hold `package-id` will be a bit cleaner, it is used in several places, and keep copy and pasting it is a bit tedious.

Reply:
Good idea. I actually did just that eventually in my environment!
Done.